### DISCUSSION TITLE
v2中能否手动指定组件type？

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#1263] [Q&A 2.x] [answered]
v2能否手动指定哪些组件是VoidField、ArrayField，而不需要通过在schema中指定type。
比如给SchemaField传一个`componentType`属性`componentType: {void：["Modal","Button"],array:["ArrayTable","List"]}`。

因为传递给SchemaField的components可能有很多自定义的组件而且数据类型是已知的，如果还需要在schema中指定一下type未免有些麻烦。

**Comment by janryWang** [comment, marked as answer]
https://react.formilyjs.org/api/shared/schema#registervoidcomponents
